tests/auth_tests/test_validators.py

```diff
@@ -259,3 +264,73 @@ def test_ascii_validator(self):
             with self.subTest(invalid=invalid):
                 with self.assertRaises(ValidationError):
                     v(invalid)
+
+    def test_username_minimum_length_validator(self):
+        valid_usernames = ['guido', 'raymond']
+        invalid_usernames = ['1t', 'tw', '']
+        v = validators.UsernameMinimumLengthValidator(min_length=3)
+        for valid in valid_usernames:
+            with self.subTest(valid=valid):
+                v(valid)
+        for invalid in invalid_usernames:
+            with self.subTest(invalid=invalid):
+                with self.assertRaises(ValidationError):
+                    v(invalid)
+
+    def test_username_validators_help_text(self):
+        validator_config = [
+            {'NAME': 'django.contrib.auth.validators.UnicodeUsernameValidator'},
+            {'NAME': 'django.contrib.auth.validators.UsernameMinimumLengthValidator', 'OPTIONS': {
+                'min_length': 3,
+            }},
+        ]
+        ASCII_and_unicode = _(
+            'Enter a valid username. This value may contain only English letters, '
+            'numbers, and @/./+/-/_ characters.'
+        )
+        length = 'Your username must contain at least 3 characters.'
+        validators_list = get_username_validators(validator_config)
+        help_texts = username_validators_help_texts(validators_list)
+        self.assertEqual(help_texts[0], ASCII_and_unicode)
+        self.assertEqual(help_texts[1], length)
+
+
+@override_settings(AUTH_USERNAME_VALIDATORS=[
```

Comment:
Why is this override applied to the entire class when you then override it with the simpler, default setting in line 311? It seems to me you could have no override here, then override only for `test_customized_validation` and that way you would be more directly checking in `test_get_default_username_validators` that the default was populated correctly. Imagine: what if somebody changes global_settings.py later, would your test for the default validators fail?

Reply:
Thanks a lot. This feedback was an eye opener. I will now think about this every time I write test.